05_classify_mutations_multimodal/nbconverted/plot_multimodal_results.py

```diff
@@ -0,0 +1,304 @@
+
+
+# ### Compare best-performing single-omics and multi-omics data types
+
+# In[10]:
+
+
+# for each data type, classify it as single-omics or multi-omics
+compare_df['model_type'] = 'single-omics'
+# multi-omics data types are concatenated using dots
+compare_df.loc[compare_df.training_data.str.contains('\.'), 'model_type'] = 'multi-omics'
+print(compare_df.training_data.unique())
+compare_df[compare_df.gene == 'TP53'].head(10)
+
+
+# In[11]:
+
+
+sns.set({'figure.figsize': (10, 6)})
+
+plot_df = pd.DataFrame()
+
+# plot mean performance over all genes in pilot experiment
+for ix, gene in enumerate(results_df.identifier.unique()):
+    
+    plot_gene_df = compare_df[(compare_df.gene == gene)].reset_index(drop=True)
+    
+    # get the best-performing data types from the single-omics and multi-omics models
+    max_single_data_type = (
+        plot_gene_df[plot_gene_df.model_type == 'single-omics']
+          .groupby('training_data')
+          .agg('mean')
+          .delta_aupr.idxmax()
+    )
+    max_multi_data_type = (
+        plot_gene_df[plot_gene_df.model_type == 'multi-omics']
+          .groupby('training_data')
+          .agg('mean')
+          .delta_aupr.idxmax()
+    )
+    
+    # get samples with that data type
+    max_single_df = plot_gene_df[plot_gene_df.training_data == max_single_data_type]
+    max_multi_df = plot_gene_df[plot_gene_df.training_data == max_multi_data_type]
+    
+    # calculate difference between means and t-test p-val for that data type
+    mean_diff = max_single_df.delta_aupr.mean() - max_multi_df.delta_aupr.mean()
+    _, p_val = ttest_ind(max_single_df.delta_aupr.values,
+                         max_multi_df.delta_aupr.values)
+    print('{} diff: {:.4f} (pval: {:.4f})'.format(gene, mean_diff, p_val))
+    
+    plot_df = pd.concat((plot_df, max_single_df, max_multi_df))
+
+sns.boxplot(data=plot_df, x='gene', y='delta_aupr', hue='model_type')
+plt.title('Best performing single-omics vs. multi-omics data type, per gene')
+plt.xlabel('Target gene')
+plt.ylabel('AUPR(signal) - AUPR(shuffled)')
+
+
+# Summary and interpretation:
+# 
+# * In this last figure, we can see that the difference in distributions between the best-performing single-omics and multi-omics model is visually quite small for all of the target genes, and none of the differences are statistically significant using a t-test over the 8 total CV folds.
+# * Looking at the other two figures, when we consider all data types the difference between single-omics and multi-omics models is still pretty minimal in general. There are some genes where adding expression data helps a lot (EGFR, PIK3CA, KRAS) and others where it doesn't really (IDH1, SETD2), but in general there's not a lot of added predictive ability when expression is combined with other data types, suggesting that methylation may be most useful in that it correlates with expression patterns (although more follow-up would be needed to really confirm this).
```

Comment:
is there any data-related differences between (EGFR, PIK3CA, KRAS) and (IDH1, SETD2)? I'm just curious if this could be driven by a confounder or if it's biologically driven. I'm mostly curious because I'm wondering if this is a dataset size issue; are EGFR, PIK3CA, KRAS more frequently mutated?

Reply:
Valid cancer types for each gene included in the train/test sets are here: https://docs.google.com/presentation/d/1pudohfOLGPa7DPIeyOIPZst-V0atzqaFdjKzuHC6neo/edit?usp=sharing

I guess the first three genes tend to be a bit more frequently mutated in some cancer types (particularly KRAS and PIK3CA), but IDH1 is very commonly mutated in LGG. Curious if you have any other thoughts based on those tables (`status_count` is the total number of mutated samples for that gene in that cancer type, and `status_proportion` is the proportion of mutated samples).

I did pick the first 3 genes and TP53 because we know they have a strong gene expression signature, and I picked IDH1 and SETD2 because they're involved in DNA methylation, so it's not exactly a random sample of genes. We could/might run this for more genes, but based on our previous results I'd expect the results to be similarly favorable to gene expression for other genes.

Reply:
Yeah, I think this makes sense. I don't have further thoughts here, seeing that the cancer types are pretty diverse in each of the predictions and that it's not super biased in all mutations (except maybe IDH1), I think it looks good.